### PR TITLE
docs(vrl): Add documentation for new `zip` function

### PR DESCRIPTION
**Pull request by bruceg**
## Summary

Ref: https://github.com/vectordotdev/vrl/issues/1157 and https://github.com/vectordotdev/vrl/pull/1158

## Change Type
- [ ] Bug fix
- [ ] New feature
- [x] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?
- [ ] Yes
- [x] No

## Does this PR include user facing changes?

- [ ] Yes. Please add a changelog fragment based on our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [ ] No. A maintainer will apply the "no-changelog" label to this PR.

## Checklist
- [ ] Please read our [Vector contributor resources](https://github.com/vectordotdev/vector/tree/master/docs#getting-started).
- [ ] If this PR introduces changes Vector dependencies (modifies `Cargo.lock`), please
  run `dd-rust-license-tool write` to regenerate the [license inventory](https://github.com/vectordotdev/vrl/blob/main/LICENSE-3rdparty.csv) and commit the changes (if any). More details [here](https://crates.io/crates/dd-rust-license-tool).

### CODE REVIEW

File: website/cue/reference/remap/functions/zip.cue [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package metadata
+
+remap: functions: zip: {
+	category: "Array"
+	description: """
+		Iterate over several arrays in parallel, producing a new array containing arrays of items from each source.
+		The resulting array will be as long as the shortest input array, with all the remaining elements dropped.
+		This function is modelled from the `zip` function [in Python](https://docs.python.org/3/library/functions.html#zip),
```
Comment (jhgilbert) [inline website/cue/reference/remap/functions/zip.cue:8]:
```suggestion
		This function is modeled from the `zip` function [in Python](https://docs.python.org/3/library/functions.html#zip),
```

**Review by jhgilbert — APPROVED** [review]
Approved with a minor suggestion for style, thanks!

**Review by pront — APPROVED** [review]

**Comment by jszwedko** [comment]
> Vector needs to update the VRL rev before this can merge. Should I include that bump in this PR or handle it separately?

https://github.com/vectordotdev/vector/pull/21936 should go in shortly at which point CI should pass on this PR.

**Comment by jszwedko** [comment]
I think it is better to update VRL separately since updating it in these docs PRs could lead to conflicts.

**Pull request merged**